### ISSUE TITLE
Boot Error on Kernel Panic

### ISSUE THREAD

**Issue by seyhadot** [tctien342/Dell-Inspiron-7591-Hackintosh#7]
i have try many clover EFI for for 7590 but i aways the same error it just how Kernel Panic Error pease visit apple.com for more infor, so what should i do now?

**Comment by tctien342** [comment]
I got same panic as u too, it cause by VoodooI2C and VoodooPS2 conflict eachother, will revert old VoodooPS2 next update

**Comment by seyhadot** [comment]
so what should i do now?

**Comment by seyhadot** [comment]
can you help me to make it working? by remote or any method to make it work? because i am new with hackintosh

**Comment by tctien342** [comment]
@seyhadot Try my current EFI
[EFI-V3-BETA.zip](https://github.com/tctien342/Dell-Inspiron-7591-Hackintosh/files/4039000/EFI-V3-BETA.zip)

**Comment by seyhadot** [comment]
it's not showing the partition of USB drive

**Comment by tctien342** [comment]
Add two files to: Clover -> drivers -> UEFI

[Archive.zip](https://github.com/tctien342/Dell-Inspiron-7591-Hackintosh/files/4039175/Archive.zip)

**Comment by seyhadot** [comment]
I use HFSPlus.efi in driver/UEFI but still not working it's the same error
like before and my laptop is 7590 4K resolution

**Comment by tctien342** [comment]
event u do this ?:
+ Goto Config.plist -> Devices -> Properties -> PciRoot(0x0)/Pci(0x2,0x0)
+ Change dpcd-max-link-rate = <14000000>

**Comment by tctien342** [comment]
Try this config!
[config.plist.zip](https://github.com/tctien342/Dell-Inspiron-7591-Hackintosh/files/4039201/config.plist.zip)

**Comment by seyhadot** [comment]
It's still not working for me it just reboot

**Comment by tctien342** [comment]
@seyhadot Try this config from 7590 4K
[config.plist.zip](https://github.com/tctien342/Dell-Inspiron-7591-Hackintosh/files/4039697/config.plist.zip)

**Comment by seyhadot** [comment]
Let me try again

**Comment by lvs1974** [comment]
Hello everybody!
I have bought Inspiron 7590 and I also tried a EFI  (OC) from this repository.
It looks very promising and safed a lot of my time.

I also have sporadic kernel panics at boot, and a reason - incompatible NVMe PCIe SSD Samsung PM991. Even when boot is successful, maximum available data transfer for this SSD is around 50-60 mbytes/sec, it is weird. I am going to replace it with a new one ([see more about non-compatible SSDs.](https://github.com/khronokernel/Anti-Hackintosh-Buyers-Guide/blob/master/Storage.md)).

I also observed some sporadic kernel panics after wake from sleep, they related to NMI and unsynced cores:
mp_kdp_enter() timed-out during locked wait after NMI;expected 12 acks but received 11 after 3636004 loops in 1296000000 ticks.

One more (quite annoying) observation: trackpad buttons sometimes do not work, I have to press them a few times to get it working.

**Comment by tctien342** [comment]
@lvs1974 download lastest master sources, will release it when test complete

**Comment by lvs1974** [comment]
@tctien342, thank you, I will give it a try.

**Comment by lvs1974** [comment]
@tctien342, I observed many crashes, where in panic report was kernel funtcion xcpm_urgency. These panics happened after wake. I have found a reason and created new Lilu plugin to solve this: https://github.com/lvs1974/CpuTscSync
Actually - it is new VoodooTSCSync, but it disables CPU power management in kernel when TSC is not in sync after wake. After syncing TSC power management   
is enabled again. Original VoodooTSCSync must be disabled or removed.
You can give it a try:
[CpuTscSync-1.2.0-RELEASE.zip](https://github.com/tctien342/Dell-Inspiron-7591-Hackintosh/files/4299954/CpuTscSync-1.2.0-RELEASE.zip)

**Comment by tctien342** [comment]
@lvs1974 , wow, thanks you for this kext
